### ISSUE TITLE
sql cli should support COPY FROM STDIN where SQL is specified with "-e" and STDIN is pure data

### ISSUE THREAD

**Issue by cucaroach** [cockroachdb/cockroach#97751]
With psql you can do this:

```
cat data.csv | psql -c "COPY FROM STDIN CSV;"
```

Where there's no SQL or start/end delimiters in the data.csv, just pure data.  I believe this is the predominate way datafiles are sent to PG using COPY from the command line.   See copyfrom roachtest for a real world example.

**Comment by sheaffej** [comment]
> I believe this is the predominate way datafiles are sent to PG using COPY from the command line.

Agreed, it's very confusing to developers that CockroachDB only supports `FROM STDIN` (and not `FROM <file.csv>`), however the data can't be read from a stdin pipe.

For posterity, one workaround I found was to prepend the data file with `COPY xxx FROM STDIN WITH ...` like so:

If the data file (file.dat) consists of:
```
HD|215000|0007862004||AA0A|C|HA|07/21/2017|10/14/2027
HD|215001|9400004290||AA0AA|A|HA|10/22/2020|12/15/2030
HD|215002|0009098482||AA0AB|A|HA|03/04/2020|03/13/2030
HD|215003|9400004484||AA0AC|E|HA|03/20/1990|03/20/2000
HD|215004|9409141023||AA0AD|A|HA|10/05/2021|10/05/2031
HD|215005|9400004670||AA0AE|E|HA|04/05/2000|03/20/2010
HD|215006|9609169987||AA0AG|E|HA|03/16/2010|03/27/2020
HD|215007|9400004934||AA0AH|E|HA|03/27/1990|03/27/2000
HD|215008|9400005006||AA0AI|A|HA|08/27/2013|11/23/2023
HD|215009|9706260116||AA0AJ|A|HA|11/06/2013|01/18/2024
```

Then the shell command would be:
```
shell> echo "COPY testtable FROM STDIN WITH CSV DELIMITER '|';" | cat - file.dat | cockroach sql <regular-connect-options>

COPY 10
```

Note the `-` first argument to `cat` to have it read the first file from stdin.

**Comment by awoods187** [comment]
@DingDLu and @rytaft PTAL and evaluate this

**Comment by rytaft** [comment]
This is an improvement suggestion to make the CLI behave like psql, since there are some docs on the web that suggest it. It currently doesn't work with CRDB without some fanfare. We don't think this is in the SQL Queries product area, but could probably take it over if no other team has bandwidth (seems like more of a CLI / SQL Foundations project...). Doesn't seem like top priority right now.

**Comment by awoods187** [comment]
I think we should be reducing friction like this wherever we see it--particularly if it impacts the onboarding or getting into production journeys. Since SQL Queries product area owns data ingestion, friction for data ingestion is within that product area--i.e., @DingDLu does own this. It's possible you mean the SQL Queries engineering team shouldn't own this which i think is debatable--it seems reasonable to me that you would do it as overall stewardship or lobby SQL foundations to do it as a customer. I do think we should eliminate this friction in 23.1, assuming the level of effort isn't significant.

**Comment by rytaft** [comment]
> I do think we should eliminate this friction in 23.1, assuming the level of effort isn't significant.

Do you mean 23.2? The SQL Queries roadmap for 23.2 is quite over-capacity at the moment, so unless we deprioritize something else I'm not sure how this will fit. It may not be a significant effort for someone who is familiar with the CLI codebase, but we don't have that expertise on the SQL Queries team, so there would be some ramp-up required.

**Comment by mgartner** [comment]
> I believe this is the predominate way datafiles are sent to PG using COPY from the command line.

Is it more common than psql's `\copy` command ([example](https://stackoverflow.com/a/30951435))?

> Agreed, it's very confusing to developers that CockroachDB only supports FROM STDIN (and not FROM <file.csv>), however the data can't be read from a stdin pipe.

Would support for `FROM file.csv` be preferred over the STDIN method?

**Comment by rytaft** [comment]
Per offline discussion, we'll try to get this done for 23.2.

**Comment by mgartner** [comment]
@DingDLu is this still a priority for v23.2?
